dask_planner/src/sql/logical/sort.rs

```diff
@@ -0,0 +1,79 @@
+use crate::expression::PyExpr;
+
+use datafusion_expr::logical_plan::Sort;
+pub use datafusion_expr::{logical_plan::LogicalPlan, Expr};
+use pyo3::prelude::*;
+
+#[pyclass(name = "Sort", module = "dask_planner", subclass)]
+#[derive(Clone)]
+pub struct PySort {
+    sort: Sort,
+}
+
+#[pymethods]
+impl PySort {
+    /// Returns a Vec of the sort expressions
+    #[pyo3(name = "getCollation")]
+    pub fn sort_expressions(&self) -> PyResult<Vec<PyExpr>> {
+        let mut sort_exprs: Vec<PyExpr> = Vec::new();
+        for expr in &self.sort.expr {
+            println!("The expr seen is {}", expr);
+            sort_exprs.push(PyExpr::from(expr.clone(), Some(self.sort.input.clone())));
+        }
+        Ok(sort_exprs)
+    }
+    #[pyo3(name = "getColumnName")]
+    pub fn get_column(&self, expr: PyExpr) -> String {
```

Comment:
Ideally I'd extended expressions.rs to handle this and have a helper utility that just returns all the column names given a vector of sort based PyExprs

Reply:
This is something that has came up a lot on my side as well. I have spoken with some folks in the DataFusion community and it is something we are going to try and get added into DataFusion itself.

Reply:
I've been running into similar issues as well, within DataFusion. I will take a look at this specific example though and see what I can do to help (either here or in DF or both).

Reply:
Ignoring error handling for now, here is my proposal. We can add the following utility in `expressions.rs` and I will create a PR for this shortly.

``` rust
/// Create a [DFField] representing an [Expr], given an input [LogicalPlan] to resolve against
pub fn expr_to_field(expr: &Expr, input_plan: &LogicalPlan) -> Result<DFField> {
    // TODO this is not the implementation that we really want and will be improved
    // once some changes are made in DataFusion
    let fields = exprlist_to_fields(&[expr.clone()], &input_plan.schema())?;
    Ok(fields[0].clone())
}
```

Then we can implement the `get_column` method as follows:

``` rust
#[pyo3(name = "getColumnName")]
pub fn get_column(&self, expr: PyExpr) -> String {
    match &expr.expr {
        Expr::Sort { expr, .. } => {
            let field = expr_to_field(expr.as_ref(), &self.sort.input).unwrap();
            field.unqualified_column().name.clone()
        },
        _ => panic!("Provided expression is not a sort expression"),
    }
}
```